### PR TITLE
Simplify SupportsConst(I|L)Div to Supports(I|L)MulHigh

### PR DESCRIPTION
**Pull request by jmesyou**
Simplify SupportsConst(I|L)Div to Supports(I|L)MulHigh

This is motivated by openj9#17861.
A check whether the underlying target supports 64 bit
multiplication returning the high word is needed
in order to intrinsify java.lang.Math.multiplyHigh

As a side effect, this has simplified the
SupportsConst(I|L)Div checks as they are transitive
to the newly added checks

Note this commit is a duplicate of PR#7082 due to a 
failed merge from a synchronized merge dependency. 
This commit differs in that getSupportsLowering(I|L)Div is retained
to avoid the same issue for the OpenJ9 downstream dependency.

When the downstream dependency is eliminated, a future
change will remove getSupportsLowering(I|L)Div

Signed-off-by: James You <james.you@protonmail.com>

### CODE REVIEW

**Comment by knn-k** [comment]
Jenkins build all,amac

File: compiler/optimizer/OMRSimplifierHandlers.cpp [hunk 1]
```diff
@@ -8690,7 +8690,7 @@ TR::Node *idivSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
                }
             node->getFirstChild()->incReferenceCount();
             }
-         else if (s->cg()->getSupportsLoweringConstIDiv() && !isPowerOf2(divisor) &&
+         else if (s->cg()->getSupportsLMulHigh() && !isPowerOf2(divisor) &&
```
Comment (knn-k) [inline compiler/optimizer/OMRSimplifierHandlers.cpp:8693]: It is `getSupportsIMulHigh()`, isn't it?

File: compiler/x/codegen/OMRCodeGenerator.cpp [hunk 2]
```diff
@@ -428,10 +428,10 @@ OMR::X86::CodeGenerator::initializeX86(TR::Compilation *comp)
    static char * enableMulHigh = feGetEnv("TR_X86MulHigh");
    if (enableMulHigh)
       {
-      self()->setSupportsLoweringConstIDiv();
+      self()->setSupportsIMulHigh();
 
       if (comp->target().is64Bit())
-         self()->setSupportsLoweringConstLDiv();
+         self()->setSupportsIMulHigh();
```
Comment (knn-k) [inline compiler/x/codegen/OMRCodeGenerator.cpp:434]: I think it should be `setSupportsLMulHigh()` here.

File: compiler/optimizer/OMRSimplifierHandlers.cpp (hunk 1 above)
Comment (jmesyou) [reply compiler/optimizer/OMRSimplifierHandlers.cpp:8693]: Yes!

**Comment by knn-k** [comment]
Jenkins build all,amac

**Review by knn-k — APPROVED** [review]

**Comment by knn-k** [comment]
This PR does not require a coordinated merge with https://github.com/eclipse-openj9/openj9/pull/17892 because `getSupportsLoweringConstIDiv()` and `getSupportsLoweringConstLDiv()` are still available.
Merging.

**Pull request merged**